activesupport/test/core_ext/hash_ext_test.rb

```diff
@@ -1762,4 +1762,10 @@ def test_expansion_count_is_limited
       Hash.from_xml(attack_xml)
     end
   end
+
+  def test_from_xml_allowed_on_frozen_input
+    assert_nothing_raised do
```

Comment:
Do we need `assert_nothing_raised`? In any case, the test will fail without `dup`.